### PR TITLE
Mysql_db single_transaction and quick

### PR DESCRIPTION
**Pull request by hyperized**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

mysql_db
##### Summary:

Adding functionality for 'single_transaction' and 'quick' to aid fast dumping in large (InnoDB) databases.
##### Example:

```
- mysql_db:
    state: dump
    name: database
    target: dump.sql.gz
    single_transaction: true
    quick: true
```

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @hyperized. To the current maintainers, @Jmainguy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by resmo** [comment]
@hyperized there are merge commits in this PR. Could you please rebase it?

**Comment by Jmainguy** [comment]
Code looks pretty good, I just need to test it. Will feedback once I have done that

**Comment by hyperized** [comment]
@resmo I'm not very familiar with rebasing, as you can see I usually merge things. Can you point me in the right direction? I know the commands, but where do I rebase onto? d9968b0, its parent, other?

**Comment by hyperized** [comment]
@resmo rebased :)

**Comment by gregdek** [comment]
Thanks @hyperized for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by hyperized** [comment]
@gregdek please review, its been rebased :)

**Comment by gregdek** [comment]
Thanks @hyperized. To the current maintainers, @Jmainguy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: database/mysql/mysql_db.py [hunk 1]
```diff
@@ -268,8 +286,7 @@ def main():
         if db == 'all':
             module.fail_json(msg="name is not allowed to equal 'all' unless state equals import, or dump.")
     try:
-        cursor = mysql_connect(module, login_user, login_password, config_file, ssl_cert, ssl_key, ssl_ca,
-                               connect_timeout=connect_timeout)
+        cursor = mysql_connect(module, login_user, login_password, config_file, ssl_cert, ssl_key, ssl_ca, connect_timeout=connect_timeout)
```
Comment (Jmainguy) [inline database/mysql/mysql_db.py:289]: looked better before, suggest changing it back

File: database/mysql/mysql_db.py [hunk 2]
```diff
@@ -328,3 +345,5 @@ def main():
 from ansible.module_utils.mysql import *
 if __name__ == '__main__':
     main()
+
+
```
Comment (Jmainguy) [inline database/mysql/mysql_db.py:349]: no need for extra blank lines, you can delete these 2

**Comment by Jmainguy** [comment]
tested and works. Make the small changes above and we can shipit

**Comment by gregdek** [comment]
Thanks again to @hyperized for this PR, and thanks @Jmainguy for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@Jmainguy don't say "shipit" when you mean "if x, then shipit". The bot isn't smart enough to understand the difference. :)

**Comment by resmo** [comment]
I see some potential problems, because these new args are booleans, they should be used as booleans:

E.g.

``` yaml
- mysql_db:
    state: dump
    name: database
    target: dump.sql.gz
    single_transaction: true
    quick: false
```

As quick is not None it would be used, this wouldn't be expected. 

See comments inline.

File: database/mysql/mysql_db.py [hunk 3]
```diff
@@ -226,6 +240,8 @@ def main():
             ssl_ca=dict(default=None),
             connect_timeout=dict(default=30, type='int'),
             config_file=dict(default="~/.my.cnf"),
+            single_transaction=dict(default=None),
+            quick=dict(default=None),
         ),
```
Comment (resmo) [inline database/mysql/mysql_db.py:245]: use bool here

``` python
single_transaction=dict(default=False, type='bool'),
quick=dict(default=False, type='bool'),
```

File: database/mysql/mysql_db.py [hunk 4]
```diff
@@ -124,6 +134,10 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
         cmd += " --all-databases"
     else:
         cmd += " %s" % pipes.quote(db_name)
+    if single_transaction is not None:
```
Comment (resmo) [inline database/mysql/mysql_db.py:137]: test for True

``` python
if single_transaction:
```

File: database/mysql/mysql_db.py [hunk 5]
```diff
@@ -124,6 +134,10 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
...
+        cmd += " --single-transaction=true"
+    if quick is not None:
```
Comment (resmo) [inline database/mysql/mysql_db.py:139]: test for True

``` python
if quick:
```

File: database/mysql/mysql_db.py [hunk 6]
```diff
@@ -56,6 +56,16 @@
       - Location, on the remote host, of the dump file to read from or write to. Uncompressed SQL
         files (C(.sql)) as well as bzip2 (C(.bz2)), gzip (C(.gz)) and xz (Added in 2.0) compressed files are supported.
     required: false
+  single_transaction:
+    description:
+      - Execute the dump in a single transaction
+    required: false
+    default: null
```
Comment (resmo) [inline database/mysql/mysql_db.py:63]: set `default: false` and also add `version_added: "2.1"`

File: database/mysql/mysql_db.py [hunk 7]
```diff
@@ -56,6 +56,16 @@
...
+  quick:
+    description:
+      - Option used for dumping large tables
+    required: false
+    default: null
```
Comment (resmo) [inline database/mysql/mysql_db.py:68]: set `default: false` and also add `version_added: "2.1"`

**Comment by hyperized** [comment]
Awesome, so no shipit after all. I'd hope for a little more consistency in the future review process. I'll try to get all fixes in today.

**Comment by resmo** [comment]
@hyperized Thanks for your feedback. I don't like it to see you unhappy how it went on this PR, my apologize.

The core and core community are working hard to continuing improving the open PR handling, but at the end it is the code and the users counts the most, to achieve a consistent ansible experience.

Thanks for your understanding, set 'ready_for_review' if the works are done, I'll handle it fast tracked.

**Comment by hyperized** [comment]
Meh, I dont know git as well as I should. Have a little patience with me, sorry.

**Comment by hyperized** [comment]
Derp, I'll make a new PR :)

**Comment by Jmainguy** [comment]
@hyperized I should have caught all the things @resmo caught during my review. So I apologize for not catching them during my review, and I will do better in future reviews. Thanks again for submitting this code.

**Pull request closed without merging**